kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java

```diff
@@ -86,4 +107,29 @@ public void testDelete() {
     assertFalse(deleted);
   }
 
+  @Test
+  public void testCreateOrReplaceWithoutDeleteExisting() {
+    server.expect().get().withPath("/api/v1/namespaces/ns1/services/my-service").andReturn(200 , service).once();
+    server.expect().get().withPath("/api/v1/namespaces/ns1/configmaps/my-configmap").andReturn(200, configMap).once();
+    server.expect().post().withPath("/api/v1/namespaces/ns1/services").andReturn(200, service).once();
+    server.expect().post().withPath("/api/v1/namespaces/ns1/configmaps").andReturn(200, configMap).once();
+
+    KubernetesClient client = server.getClient();
+    KubernetesList list = new KubernetesListBuilder().withItems(service, configMap).build();
+    client.resourceList(list).inNamespace("ns1").createOrReplace();
```

Comment:
I think that we also need to verify that the server did got everything it expected.